New package: rocksdb-5.17.2

magenbluten opened the pull request:
This provides a RocksDB package. Basically this is just facebook's fork and mods to google's LevelDB.

abenson commented on srcpkgs/rocksdb/template:
```diff
@@ -0,0 +1,33 @@
+	vcopy "include/rocksdb/*" usr/include/rocksdb
+	vlicense LICENSE.Apache
+	vlicense LICENSE.leveldb
+	vlicense COPYING
```
You don't need to include the GPL

abenson commented on srcpkgs/rocksdb/template:
```diff
@@ -0,0 +1,33 @@
+configure_args="-DCMAKE_BUILD_TYPE=Release -DWITH_TESTS=0 -DUSE_RTTI=0 -DWITH_LZ4=1"
+hostmakedepends="git"
+makedepends="liblz4-devel"
+depends="liblz4"
```
Shouldn't be necessary if it links to this library, just need the -devel in makedepends

abenson commented on common/shlibs:
```diff
@@ -3396,3 +3396,4 @@ libmp3splt.so.0 libmp3splt-0.9.2_1
 libliquid.so.1 liquid-dsp-1.3.1_1
 libsnore-qt5.so.0.7 snorenotify-0.7.0_1
 libsnoresettings-qt5.so.0.7 snorenotify-0.7.0_1
+librocksdb.so.5 rocksdb-5.17.2_1
```
Missing newline at end of line

abenson commented on srcpkgs/rocksdb/template:
```diff
@@ -0,0 +1,33 @@
+revision=1
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE=Release -DWITH_TESTS=0 -DUSE_RTTI=0 -DWITH_LZ4=1"
+hostmakedepends="git"
```
What does it need git for?

magenbluten replied:
actually, it was a warning at "configure" time. i'll remove it.

Johnnynator commented on srcpkgs/rocksdb/template:
```diff
@@ -0,0 +1,33 @@
+version=5.17.2
+revision=1
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE=Release -DWITH_TESTS=0 -DUSE_RTTI=0 -DWITH_LZ4=1"
```
`option(PORTABLE "build a portable binary" OFF)`
set `-DPORTABLE=1`, we don't want to build with `march=native`

magenbluten replied:
will add the flag, hopefully have time tomorrow

maxice8 commented on common/shlibs:
```diff
@@ -3396,3 +3396,5 @@ libmp3splt.so.0 libmp3splt-0.9.2_1
 libsnore-qt5.so.0.7 snorenotify-0.7.0_1
 libsnoresettings-qt5.so.0.7 snorenotify-0.7.0_1
+librocksdb.so.5 rocksdb-5.17.2_1
+
```
extra line, remove

maxice8 replied:
remove the extra line at the end of the file, it is not meant to be there

maxice8 commented on srcpkgs/rocksdb/template:
```diff
@@ -0,0 +1,30 @@
+makedepends="liblz4-devel"
+short_desc="RocksDB is a embeddable, persistent key-value store for fast storage"
+maintainer="magenbluten <mb0@codemonkey.cc>"
+license="GPL-2.0-or-later,Apache-2"
```
need a whitespace between the coma and Apache-2

maxice8 replied:
Apache-2 is not an SPDX identifier, Apache-2.0 is problably what you want

maxice8 commented on srcpkgs/rocksdb/template:
```diff
@@ -0,0 +1,30 @@
+
+post_install() {
+	vcopy "include/rocksdb/*" usr/include/rocksdb
+	vlicense LICENSE.Apache
```
Don't need to vlicense for Apache

maxice8 commented on srcpkgs/rocksdb/template:
```diff
@@ -0,0 +1,30 @@
+makedepends="liblz4-devel"
+short_desc="RocksDB is a embeddable, persistent key-value store for fast storage"
+maintainer="magenbluten <mb0@codemonkey.cc>"
+license="GPL-2.0-or-later,Apache-2"
```
you are including LICENSE.leveldb which is BSD-3-Clause, it should also be in the license file.

magenbluten replied:
oh yes. actually i think facebook totally screwed the licenses. i did not include it because the readme says you can choose between gpl or apache-2 at will:

RocksDB is dual-licensed under both the GPLv2 (found in the COPYING file in the root directory) and Apache 2.0 License (found in the LICENSE.Apache file in the root directory). You may select, at your option, one of the above-listed licenses.

maxice8 commented:
On Fri, Feb 08, 2019 at 11:54:11AM -0800, mb0 wrote:
> magenbluten commented on this pull request.
> 
> 
> 
> > @@ -0,0 +1,30 @@
> +# Template file for 'rocksdb'
> +pkgname=rocksdb
> +version=5.17.2
> +revision=1
> +build_style=cmake
> +configure_args="-DCMAKE_BUILD_TYPE=Release -DPORTABLE=1 -DWITH_TESTS=0 -DUSE_RTTI=0 -DWITH_LZ4=1"
> +makedepends="liblz4-devel"
> +short_desc="RocksDB is a embeddable, persistent key-value store for fast storage"
> +maintainer="magenbluten <mb0@codemonkey.cc>"
> +license="GPL-2.0-or-later,Apache-2"
> 
> oh yes. actually i think facebook totally screwed the licenses. i did not include it because the readme says you can choose between gpl or apache-2 at will:
> 
> RocksDB is dual-licensed under both the GPLv2 (found in the COPYING file in the root directory) and Apache 2.0 License (found in the LICENSE.Apache file in the root directory). You may select, at your option, one of the above-listed licenses.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/8204#discussion_r255215759

Well you are the maintainer you pick

magenbluten commented:
> On Fri, Feb 08, 2019 at 11:54:11AM -0800, mb0 wrote: magenbluten commented on this pull request. > @@ -0,0 +1,30 @@ +# Template file for 'rocksdb' +pkgname=rocksdb +version=5.17.2 +revision=1 +build_style=cmake +configure_args="-DCMAKE_BUILD_TYPE=Release -DPORTABLE=1 -DWITH_TESTS=0 -DUSE_RTTI=0 -DWITH_LZ4=1" +makedepends="liblz4-devel" +short_desc="RocksDB is a embeddable, persistent key-value store for fast storage" +maintainer="magenbluten ***@***.***>" +license="GPL-2.0-or-later,Apache-2" oh yes. actually i think facebook totally screwed the licenses. i did not include it because the readme says you can choose between gpl or apache-2 at will: RocksDB is dual-licensed under both the GPLv2 (found in the COPYING file in the root directory) and Apache 2.0 License (found in the LICENSE.Apache file in the root directory). You may select, at your option, one of the above-listed licenses. -- You are receiving this because you commented. Reply to this email directly or view it on GitHub: [#8204 (comment)](https://github.com/void-linux/void-packages/pull/8204#discussion_r255215759)
> Well you are the maintainer you pick

so i piked a license meaning i removed the leveldb reference. i addressed all other comments. will you integrate the package? thx.

maxice8 commented:
Please rebase your changes to fix conflicts

maxice8 commented:
Looking at the [Travis logs](https://travis-ci.org/void-linux/void-packages/jobs/493423255) it seems atomic is missing from armv6l-musl.

Please add the following lines to the template:

```
case "$XBPS_TARGET_MACHINE" in
    armv[56]*) 
    makedepends+=" libatomic-devel"
    LDFLAGS+=" -latomic"
    ;;
esac
```

magenbluten commented:
@maxice8 thanks for helping me out!

i've seen you bumped the version to 5.18.3, the latest official release is still 5.17.2. it's fine with me. thanks again!
